component/api/src/main/java/org/exoplatform/social/common/lifecycle/LifeCycleCompletionService.java

```diff
@@ -75,17 +77,11 @@ public LifeCycleCompletionService(InitParams params) {
       this.configAsyncExecution = DEFAULT_ASYNC_EXECUTION;
     }
 
+    this.asynchronousExecutor = Executors.newFixedThreadPool(this.configThreadNumber);
 
-    //
-    if (configAsyncExecution) {
-      this.executor = Executors.newFixedThreadPool(this.configThreadNumber);
-    }
-    else {
-      this.executor = new DirectExecutor();
-    }
+    this.synchronousExecutor = new DirectExecutor();
 
-    //
-    this.ecs = new ExecutorCompletionService(executor);
+    this.ecs = new ExecutorCompletionService(asynchronousExecutor);
```

Comment:
Im not sure to understand here. As tasks are added with ecs.submit, I m afraid taht all task will be added as async.
But we want only listener with asynchronous annotation. Other must still be with DirectExectutor, no ?

Reply:
addTask is used by the AbstractLifeCycle only when the listener has the asynchronous annotation to create a new thread and dispatch the event. In the case of an asynch listener, AbstractLifeCycle will dispatch the event without creating a new thread, so it will be executed in the same thread.

Reply:
okWhat if lifecycleCompletionService is configured as async (async=true) ? 
Should we use direct executor ? 
Should we use asyncexectutor only with async=true AND Asynchronous annotation ?

Reply:
> what if lifecycleCompletionService is configured as async (async=true) ? 
if async is true all listeners will be asynchronous
> Should we use direct executor ? Should we use asyncexectutor only with async=true AND Asynchronous annotation ?
fyi I changed the code to only use Executors.newFixedThreadPool(), we don't need the directExecutor anymore.
so in case the listener is async, a new thread will be created; otherwise, the code will be executed in the same thread.

Reply:
Let me understand :

Can you complete theses cases : 
- async = true : 
   - listener with @Asynchronous : same thread or new thread ?
   - listener without @Asynchronous : same thread or new thread ?
- async = false : 
   - listener with @Asynchronous : same thread or new thread ?
   - listener without @Asynchronous : same thread or new thread ?

Reply:
> Let me understand :
> 
> Can you complete theses cases :
> 
> * async = true :
>   
>   * listener with @asynchronous : same thread or new thread ?
New Thread
>   * listener without @asynchronous : same thread or new thread ?
New Thread
> * async = false :
>   
>   * listener with @asynchronous : same thread or new thread ?
New Thread
>   * listener without @asynchronous : same thread or new thread ?
Same Thread

Reply:
ok for this part